src/controllers/ConfirmationController.ts

```diff
@@ -1,19 +1,19 @@
 import { controller, httpGet, BaseHttpController } from 'inversify-express-utils';
 import { CONFIRMATION_PAGE_URI } from '../utils/Paths';
+import { Request } from 'express';
+import { SessionMiddleware } from 'ch-node-session-handler';
+import { AuthMiddleware } from '../middleware/AuthMiddleware';
 
-@controller(CONFIRMATION_PAGE_URI)
+@controller(CONFIRMATION_PAGE_URI, SessionMiddleware, AuthMiddleware)
 export class ConfirmationController extends BaseHttpController {
 
     @httpGet('')
-    public getConfirmationView(): void {
+    public getConfirmationView(req: Request): void {
 
-        const session = this.httpContext.request.session;
-        let companyNumber: string = '';
+        req.session.chain(_ => _.getExtraData())
+            .map(data => data.appeals.penaltyIdentifier.companyNumber)
+            .map(companyNumber => this.httpContext.response.render('confirmation', { companyNumber }))
```

Comment:
It would be ideal if we did not have to make two render calls and structure code in a way so that it reads session first and then it renders.